Rest.Services.Test/ProductRepositoryTest.cs

```diff
@@ -0,0 +1,84 @@
+using Xunit;
+using Rest.Models;
+using Rest.Services;
+using System.Collections.Generic;
+using System;
+
+namespace Rest.Services.Test;
+
+public class ProductRepositoryTest : IDisposable
+{
+    ProductRepository _sut;
+    List<Product> _products;
+
+    public ProductRepositoryTest()
+    {
+        _sut = new ProductRepository();
+        _products = new List<Product>();

```

Comment:
You can't use a global resource like a list of data to use it in your test. If something change, could affect one or many test.

If you need to **simulate** a data store, Moking is way to do it.

For `.NET Core` there are a few good resources like `Moq`, `FakeitEasy`, `NSubstitute`, etc. Feel free to explore the solution you like the most.

Reply:
You have a session yo explain how it works

Reply:
Done.